packages/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol

```diff
@@ -13,50 +13,64 @@ import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
 ///         for sending and receiving data on the L1 side. Users are encouraged to use this
 ///         interface instead of interacting with lower-level contracts directly.
 contract L1CrossDomainMessenger is CrossDomainMessenger, ISemver {
-    /// @notice Address of the OptimismPortal. This will be removed in the
-    ///         future, use `portal` instead.
-    /// @custom:legacy
-    OptimismPortal public immutable PORTAL;
-
-    /// @notice Address of the SuperchainConfig contract.
+    /// @notice Contract of the SuperchainConfig.
     SuperchainConfig public superchainConfig;
 
     /// @notice Semantic version.
-    /// @custom:semver 2.1.1
-    string public constant version = "2.1.1";
+    /// @custom:semver 2.2.0
+    string public constant version = "2.2.0";
+
+    /// @notice Address of the OptimismPortal contract.
+    /// @custom:network-specific
+    address payable public portal;
 
     /// @notice Constructs the L1CrossDomainMessenger contract.
-    /// @param _portal Address of the OptimismPortal contract on this network.
-    constructor(OptimismPortal _portal) CrossDomainMessenger(Predeploys.L2_CROSS_DOMAIN_MESSENGER) {
-        PORTAL = _portal;
-        initialize({ _superchainConfig: SuperchainConfig(address(0)) });
+    constructor() CrossDomainMessenger() {
+        initialize({
+            _superchainConfig: SuperchainConfig(address(0)),
+            _portal: payable(address(0)),
+            _l2CrossDomainMessenger: Predeploys.L2_CROSS_DOMAIN_MESSENGER
+        });
     }
 
     /// @notice Initializes the contract.
-    /// @param _superchainConfig Address of the SuperchainConfig contract on this network.
-    function initialize(SuperchainConfig _superchainConfig) public initializer {
+    /// @param _superchainConfig Contract of the SuperchainConfig on this network.
+    /// @param _portal Address of the OptimismPortal contract on this network.
+    /// @param _l2CrossDomainMessenger Address of the L2CrossDomainMessenger contract on the other network.
+    function initialize(
+        SuperchainConfig _superchainConfig,
+        address payable _portal,
+        address _l2CrossDomainMessenger
+    )
+        public
+        initializer
+    {
         superchainConfig = _superchainConfig;
-        __CrossDomainMessenger_init();
+        portal = _portal;
+        __CrossDomainMessenger_init({ _otherMessenger: _l2CrossDomainMessenger });
     }
 
-    /// @notice Getter for the OptimismPortal address.
-    function portal() external view returns (address) {
-        return address(PORTAL);
+    /// @notice Getter function for the OptimismPortal contract on this chain.
+    ///         Public getter is legacy and will be removed in the future. Use `portal()` instead.
+    /// @return Contract of the OptimismPortal on this chain.
+    /// @custom:legacy
+    function PORTAL() external view returns (OptimismPortal) {
+        return OptimismPortal(portal);
```

Comment:
this should be able to just return `portal` if we update the type in the storage